### DISCUSSION TITLE
Is it possible to have multiple Commander PROs and use the temperature sensor from one of them to control the RPM of all the fans spread over all the Commander PROs?

### DISCUSSION THREAD

**Discussion by Mannekino** [liquidctl/liquidctl#496] [Q&A] [answered]
Hello,

I'm working on a build soon that currently has three Commander PROs but I'm looking to reduce this to two.

It's a custom loop water cooled system and I would like to control the fan RPM based on the temperature of the coolant in the loop. I'm using a single temperature probe in the loop and I want to hook that up to one of the Commander PROs.

Can I control the RPM of the fans hooked up to all the Commander PROs based on a single temperature sensor hooked up to one of the Commander PROs? There will be 13 fans in the system in total. How would I need to modify the command below to achieve this?

`liquidctl set fan1 speed 20 500 40 1500`

How do I specify which Commander PROs I'm addressing and which temperature probe to use for the source?

If this is not possible I would probably need to hook up a temperature probe to each Commander PRO, is that correct?

Thank you for your help.

Kind regards,

Mannekino

**Comment by jonasmalacofilho** [comment, marked as answer]
Hi!

> Can I control the RPM of the fans hooked up to all the Commander PROs based on a single temperature sensor hooked up to one of the Commander PROs?

Yes, but not in hardware.  You'll need a script or program that periodically polls the sensor, and adjusts all fan speeds accordingly.

`extra/yoda` might be a starting point, but you'll probably need to modify it a bit for this use case.

> How do I specify which Commander PROs I'm addressing

There's a `--pick` option, to disambiguate between `--match` matches.  You can also filters like `--adress` or `--serial`.

Take a look at the output from `list --verbose` and see which options make sense.  You can find all options with `--help`, or in the man page (`man liquidctl` on Linux/Mac/BSDs, or [online](https://man.archlinux.org/man/community/liquidctl/liquidctl.8.en)).

> and which temperature probe to use for the source?

That's only possible within the same device (using the `--temperature-sensor` option).

> [...] hook up a temperature probe to each Commander PRO, is that correct?

That's _another_ way to achieve what you want.

**Comment by Mannekino** [comment]
Thank you so much for your quick reply.

I was breaking my head over how this fan curve system worked because I couldn't find a liquidctl daemon running. I figured it was a feature of the software that read the value of the temperature sensor based on a certain interval and then adjusted the RPM. But that command actually programs/configures the Commander PRO device and needs to be entered every time the system reboots if I'm understanding you correctly (which can be automated with a script). This computer has Debian 11 as its operating system so I think I can use that systemd example from the documentation.

I currently don't have the computer I'm going to work on at my location. I have test setup now with just a single Commander PRO. I do have a second Commander PRO in my NAS I could temporarily use for testing. I'm going to take that one out and expand the test setup with that. I don't know how to write Python nor am I that proficient in scripting but I will give it a shot.

I will order a second temperature probe just in case I need to stick with the hardware option. I don't mind this option but it's not the most fancy one.

**Reply by jonasmalacofilho** [reply]
> But that command actually programs/configures the Commander PRO device

Yes.

> and needs to be entered every time the system reboots if I'm understanding you correctly (which can be automated with a script).

Yes, but (in general) it should also be executed when the system resumes from a suspended state, as some devices may reset when suspended (I'm not sure how the Commander Pro behaves in this regard).

> [...] in case I need to stick with the hardware option. I don't mind this option but it's not the most fancy one.

While more crude, a benefit of the hardware-based alternative is that it's intrinsically more reliable than a software-only solution: it should[^1] continue to work even if the CPU or USB bus are saturated, or if the system crashes.

[^1]: Assuming that the embedded controller isn't very silly or buggy.

**Comment by Mannekino** [comment]
Thank you. I will stick with the hardware solution and just put a second inline temperature sensor in the loop. This isn't a big deal. There is also a time issue with the job so I don't really have the time to figure out the software/script solution. I will make a systemd startup script today and test it out and use your suggestion to properly identify each Commander PRO.
